pwiz_tools/Skyline/Controls/SmallMoleculeColumnsManager.cs

```diff
@@ -0,0 +1,307 @@
+﻿/*
+ * Original author: Brian Pratt <bspratt .at. u.washington.edu>,
+ *                  MacCoss Lab, Department of Genome Sciences, UW
+ *
+ * Copyright 2020 University of Washington - Seattle, WA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Forms;
+using pwiz.Skyline.Model;
+using pwiz.Skyline.Model.Lib;
+using pwiz.Skyline.Properties;
+
+namespace pwiz.Skyline.Controls
+{
+
+    /// <summary>
+    /// Interface which guarantees automatic management of small molecule details in
+    /// data grids originally designed for peptides
+    /// </summary>
+    public interface ISmallMoleculeColumnsManagementProvider
```

Comment:
Maybe the SmallMoleculeColumnsManager should have a TargetResolver on it, and then you would not need this interface.

Reply:
Here's a screenshot from the iRT tutorial running with the UI set to Mixed mode:
![image](https://user-images.githubusercontent.com/1200761/84829723-0bdeb780-afdd-11ea-836c-ca5f1f1c664a.png)

Reply:
and I'll see about eliminating that interface